### PR TITLE
Jira rewrite and fixes

### PR DESCRIPTION
**Pull request by orenzohar**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/15907
related: https://github.com/demisto/etc/issues/15372

## Description
- Rewritten into python
- Fetch fixed
- Improved JQL support
- getAttachments flag in `jira-get-issue`

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (https://github.com/demisto/etc/issues/14320)
- [x] Code Review

### CODE REVIEW

**Comment by yuvalbenshalom** [comment]
@orenzohar lets have a talk tomorrow
We should test this with simulating content update when the current jira integration is working and fetching incidents.

File: Integrations/Jira/Jira.py [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+            url = 'rest/api/latest/issue/{issue_id}/transitions?expand=transitions.fields'.format(issue_id=issue_id)
+            json_body = {"transition": {"id": str(j_res.get('transitions')[i].get('id'))}}
+            return jira_req('POST', url, json.dumps(json_body))
+
+    return_error('Status "{}" not found. \n'
+                 'Valid transitions are: {} \n'.format(status, transitions))
+
+
+@logger
+def get_comments_command(issue_id):
```
Comment (yuvalbenshalom) [inline Integrations/Jira/Jira.py]: this is a general change request to all of the command functions.
the `foo_command()` function should not have any arguments.
It should extract all of the relevant argents in the beginning, call a business function that will return a raw response,
and then the `foo_command()` take the response and create context outputs and human readable.(possibly with another helper function)

please see the code convention documentation for command functions and fix accordingly:
https://github.com/demisto/content/blob/master/docs/code_conventions/README.MD#command-functions

File: Integrations/Jira/Jira.py [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+''' COMMANDS MANAGER / SWITCH PANEL '''
+demisto.debug('Command being called is %s' % (demisto.command()))
+try:
+    if demisto.command() == 'test-module':
+        # This is the call made when pressing the integration test button.
+        test_module()
+
+    elif demisto.command() == 'fetch-incidents':
+        # Set and define the fetch incidents command to run after activated via integration settings.
+        fetch_incidents(**snakify(demisto.params()))
```
Comment (yuvalbenshalom) [inline Integrations/Jira/Jira.py]: after command functions will be refactored, they should not receive the args.

**Comment by glicht** [comment]
@ronykoz I've put in a fix to re-run flake8 with python3 if it fails with python2. See the commit: https://github.com/demisto/content/pull/3209/commits/2e14c76bbf488e04c6ac98b020428859c021b7f4

File: Integrations/integration-jira.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ commonfields:
   id: jira
   version: -1
 name: jira
-display: Atlassian Jira
```
Comment (yuvalbenshalom) [inline Integrations/integration-jira.yml]: this is correct, but you should mention in every command description that it is deprecated.

**Comment by yuvalbenshalom** [comment]
@kirbles19 documentation is ready for your review

**Pull request merged**